### PR TITLE
mail_print module

### PR DESCRIPTION
**Pull request by rami-wafaie**
This module makes pdf reports from the Emails, so the user will be able to print them

### CODE REVIEW

File: mail_print/README.rst [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+==========
+Mail Print
+==========
+
+This module makes pdf reports from the Emails, so the user will be able to print them.
+
+Usage
+=====
+
+* go to ...
```
Comment (rafaelbn) [inline mail_print/README.rst]: Where is the option to print?

**Comment by rami-wafaie** [comment]
@rafaelbn I added to README file where to find the option to print

**Comment by rafaelbn** [comment]
In this way, only admin usar can print an email. Have you think about user?

**Comment by OSguard** [comment]
@rafaelbn it works if you install it with mail_read_new_window
but having there a button is a nice improvment

File: mail_print/views/mail_message_report.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                            <style>
+                                .table1 {width: 50%}
+                                .table2 {width: 100%}
+                                .font {font-family: Helvetica, sans-serif;font-size: 10pt}
+                                .bold {font-weight: bold}
+                                .space {padding-right: 0.3em;}
+                            </style>
+                            <table class="table1 table-condensed">
+                                <tr>
+                                    <td class="font bold"><span>From:</span></td>
```
Comment (hbrunn) [inline mail_print/views/mail_message_report.xml]: consider using the `th` tag is you want to signify some cell is a column/row header

File: mail_print/views/mail_message_report.xml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        <template id="mail_message">
+            <t t-call="report.html_container">
+                <t t-foreach="docs" t-as="o">
+                    <t t-call="report.internal_layout">
+                        <div class="page">
+                            <style>
+                                .table1 {width: 50%}
+                                .table2 {width: 100%}
+                                .font {font-family: Helvetica, sans-serif;font-size: 10pt}
```
Comment (hbrunn) [inline mail_print/views/mail_message_report.xml]: setting a fixed size here but not touching it for everything else can lead to weird effects depending on the browser settings. What is you rationale to pick a font and size in the first place?

File: mail_print/views/mail_message_report.xml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                                <tr>
+                                    <td class="font bold"><span>To:</span></td>
+                                    <td class="font" t-foreach="o.partner_ids" t-as="p"><span t-field="p.name"/> <t t-if="p.email"> <u><span t-field="p.email"></span></u></t></td>
+                                </tr>
+                                <tr>
+                                    <td class="font bold"><span>Subject:</span></td>
+                                    <td class="font"><span t-field="o.subject"/></td>
+                                </tr>
+                                <tr>
+                                    <td class="font bold"><span>Date:</span></td>
```
Comment (hbrunn) [inline mail_print/views/mail_message_report.xml]: make al of the bold cells `th`

**Comment by hbrunn** [comment]
this is nice! But https://github.com/OCA/social/pull/69#issuecomment-226554160 would indeed be great. You can inject the button via qweb into https://github.com/OCA/OCB/blob/8.0/addons/mail/static/src/xml/mail.xml#L251, then override https://github.com/OCA/OCB/blob/8.0/addons/mail/static/src/js/mail.js#L930 to attach a handler, and call `this.do_action($your_report_action)` there. Or call https://github.com/OCA/OCB/blob/8.0/addons/web/controllers/main.py#L1236 with your action's xmlid and pass that to `do_action`

**Comment by rami-wafaie** [comment]
@hbrunn Thanks for your comments, will be done in the next days

**Comment by rami-wafaie** [comment]
Print option is added - Review comments are done, Please have a look

print-option

**Comment by hbrunn** [comment]
:+1:

**Comment by rafaelbn** [comment]
:+1:

**Pull request merged**